### PR TITLE
Basic profile

### PR DESCRIPTION
**Pull request by Sjmarf**
Re-implemented part of the profile page. Most of this is copy/paste from v1. User content and flairs aren't displayed yet, and there aren't yet any actions in the ellipsis menu.

### CODE REVIEW

File: Mlem/App/Utility/Extensions/Date+Formatter.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
+//
+//  Date+RelativeTime.swift
```
Comment (EricBAndrews) [inline Mlem/App/Utility/Extensions/Date+Formatter.swift:2]: `+Extensions` seems to be the de facto standard at this point

File: Mlem/App/Utility/Extensions/Int+abbreviated.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
+//
+//  Int+Abbreviated.swift
```
Comment (EricBAndrews) [inline Mlem/App/Utility/Extensions/Int+abbreviated.swift:2]: `+Extensions`

File: Mlem/App/Views/Pages/PersonView.swift [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            }
+            .navigationTitle(isAtTop ? "" : (person.displayName_ ?? person.name))
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .topBarTrailing) {
+                    // TODO:
+                    ToolbarEllipsisMenu {}
+                }
+            }
+        }
```
Comment (EricBAndrews) [inline Mlem/App/Views/Pages/PersonView.swift:55]: Extracting this to a `content()` function makes the `ContentLoader` structure a little prettier

File: Mlem/App/Views/Pages/PersonView.swift [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                            Divider()
+                            ProgressView()
+                                .padding(.top)
+                        }
+                        .transition(.opacity)
+                    }
+                }
+                .animation(.easeOut(duration: 0.2), value: person is any Person3Providing)
+            }
+            .navigationTitle(isAtTop ? "" : (person.displayName_ ?? person.name))
```
Comment (EricBAndrews) [inline Mlem/App/Views/Pages/PersonView.swift:47]: Potentially OOS for this PR, but we use this little `displayName_ ?? name` construction a few places--might be worth defining a computed var for convenience

File: Mlem/App/Views/Root/Tabs/Profile/Profile View.swift [hunk 5]
```diff
@@ -14,19 +15,10 @@ struct ProfileView: View {
...
-    }
-    
-    var content: some View {
-        ScrollView {
-            Markdown(markdown)
-                .padding()
+        Group {
```
Comment (EricBAndrews) [inline Mlem/App/Views/Root/Tabs/Profile/Profile View.swift:18]: Why is this in `Group`?

File: Mlem/App/Views/Shared/Avatar/ProfileHeaderView.swift [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        self.type = type
+    }
+    
+    var body: some View {
+        VStack(spacing: AppConstants.standardSpacing) {
+            AvatarBannerView(profilable, type: type)
+            Button {
+                (profilable as? any CommunityOrPersonStub)?.copyFullNameWithPrefix()
+            } label: {
+                VStack(spacing: 5) {
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/Avatar/ProfileHeaderView.swift:38]:
```suggestion
                VStack(spacing: AppConstants.halfSpacing) {
```

File: Mlem/App/Views/Shared/Avatar/ProfileHeaderView.swift [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+                    Text(profilable?.displayName_ ?? profilable?.name ?? "")
+                        .font(.title)
+                        .fontWeight(.semibold)
+                        .lineLimit(1)
+                        .minimumScaleFactor(0.01)
+                    Text(subtitle)
+                        .font(.caption)
+                        .foregroundStyle(.secondary)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/Avatar/ProfileHeaderView.swift:46]:
```suggestion
                        .foregroundStyle(palette.secondary)
```

File: Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+            if dividers.contains(.top) {
+                Divider()
+            }
+            
+            ScrollViewReader { scrollProxy in
+                ScrollView(.horizontal) {
+                    buttonStack(scrollProxy: scrollProxy, isSelected: false)
+                        .overlay {
+                            buttonStack(isSelected: true)
+                                .background(.blue)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift:62]:
```suggestion
                                .background(palette.accent)
```

File: Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    func bubbleButtonLabel(
+        tab: Value,
+        isSelected: Bool
+    ) -> some View {
+        AnyView(HStack(spacing: 8) {
+            let value = value(tab)
+            Text(label(tab))
+                .font(.subheadline)
+                .fontWeight(.semibold)
+                .foregroundColor(isSelected ? .white : .primary)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift:146]:
```suggestion
                .foregroundColor(isSelected ? .white : palette.primary)
```

File: Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+            if let value {
+                Text(value.abbreviated)
+                    .monospacedDigit()
+                    .font(.subheadline)
+                    .fontWeight(.medium)
+                    .foregroundColor(isSelected ? .white.opacity(0.8) : .secondary)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift:152]:
```suggestion
                    .foregroundColor(isSelected ? .white.opacity(0.8) : palette.secondary)
```

File: Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+    func bubbleButtonLabel2(
+        tab: Value,
+        isSelected: Bool
+    ) -> some View {
+        AnyView(HStack {
+            let value = value(tab)
+            Text(label(tab))
+                .font(.subheadline)
+                .fontWeight(.semibold)
+                .foregroundColor(isSelected ? .white : .primary)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift:170]:
```suggestion
                .foregroundColor(isSelected ? .white : palette.primary)
```

File: Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+            // .padding(value == nil ? .horizontal : .leading, 22
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift:171]: Dead code?

File: Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+            if let value {
+                Text(value.abbreviated)
+                    .monospacedDigit()
+                    .font(.subheadline)
+                    .fontWeight(.medium)
+                    .foregroundColor(isSelected ? .white.opacity(0.8) : .secondary)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift:177]:
```suggestion
                    .foregroundColor(isSelected ? .white.opacity(0.8) : palette.secondary)
```

File: Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+                    .background(
+                        Group {
+                            if value < 10 {
+                                Circle()
+                            } else {
+                                Capsule()
+                            }
+                        }
+                        .foregroundStyle(
+                            isSelected ? Color(uiColor: .systemBackground).opacity(0.3) : Color(uiColor: .secondarySystemBackground)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/BubblePicker/BubblePicker.swift:190]:
```suggestion
                            isSelected ? palette.background.opacity(0.3) : palette.secondaryBackground
```

File: Mlem/App/Views/Shared/MarkdownImageView.swift [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+
+struct MarkdownImageView: View {
+    @Environment(Palette.self) var palette
+    
+    let image: InlineImage
+    
+    var body: some View {
+        Image(systemName: "photo")
+            .imageScale(.large)
+            .foregroundStyle(.secondary)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/MarkdownImageView.swift:19]:
```suggestion
            .foregroundStyle(palette.secondary)
```

File: Mlem/App/Views/Shared/ProfileDateView.swift [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+import SwiftUI
+
+struct ProfileDateView: View {
+    var profilable: any Profile2Providing
+    
+    var body: some View {
+        HStack {
+            Label(format(profilable.created), systemImage: Icons.cakeDay)
+        }
+        .foregroundStyle(.secondary)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/ProfileDateView.swift:18]:
```suggestion
        .foregroundStyle(palette.secondary)
```

File: Mlem/App/Views/Pages/PersonView.swift [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+    }
+    
+    @ViewBuilder
+    func bio(person: any Person) -> some View {
+        if let bio = person.description_ {
+            Divider()
+            VStack(spacing: AppConstants.standardSpacing) {
+                let blocks: [BlockNode] = .init(bio)
+                if blocks.isSimpleParagraphs, bio.count < 300 {
+                    MarkdownText(blocks, configuration: .default)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Pages/PersonView.swift:65]: Build fails for me on this line: `Cannot find MarkdownText in scope`

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Code looks good--bunch of `palette` conformance comments and a couple nits.

Can't build locally--`MarkdownText` in `PersonView:65` throws an error

**Comment by Sjmarf** [comment]
> Can't build locally--MarkdownText in PersonView:65 throws an error

[I added `MarkdownText` to LemmyMarkdownUI yesterday](https://github.com/mlemgroup/LemmyMarkdownUI/pull/12). Maybe that package needs to be properly versioned for it to update automatically 🤔 Right-clicking on LemmyMarkdownUI in the sidebar shows an option to update manually

**Comment by EricBAndrews** [comment]
> > Can't build locally--MarkdownText in PersonView:65 throws an error
> 
> [I added `MarkdownText` to LemmyMarkdownUI yesterday](https://github.com/mlemgroup/LemmyMarkdownUI/pull/12). Maybe that package needs to be properly versioned for it to update automatically 🤔 Right-clicking on LemmyMarkdownUI in the sidebar shows an option to update manually

Hmm, yeah, that did it. Weirdly it didn't work if I did `Reset package caches`, which also forces a fetch 🤔 🤔 🤔

File: Mlem/App/Views/Pages/PersonView.swift (hunk 4 above)
Comment (Sjmarf) [reply Mlem/App/Views/Pages/PersonView.swift]: Yeah, that would be useful. The `displayName` property of `ApiPersonView` is optional - it's nil coalesced with `name` when `Person1` is instantiated to make `Person1.displayName` non-optional. Potentially we could make `displayName_` non-optional too, and just return `name` when the display name isn't available

**Review by EricBAndrews — APPROVED** [review]
LGTM 🔥

**Pull request merged**